sljit_src/sljitLir.h

```diff
@@ -1935,6 +1935,52 @@ SLJIT_API_FUNC_ATTRIBUTE void sljit_set_function_context(void** func_ptr, struct
 SLJIT_API_FUNC_ATTRIBUTE void sljit_free_unused_memory_exec(void);
 #endif
 
+#define SLJIT_ATOMIC_BASE 0x0
+/* Atomic load loads data from memory to a register atomically,
+   marking the memory location as locked if necessary. */
+
+/* Flags: - (does not modify flags) */
+#define SLJIT_ATOMIC_LOAD (SLJIT_ATOMIC_BASE + 1)
+
+/* Flags: - (does not modify flags) */
+#define SLJIT_ATOMIC_LOAD_U32 (SLJIT_ATOMIC_BASE + 2)
+/* Flags: - (does not modify flags) */
+#define SLJIT_ATOMIC_LOAD32 (SLJIT_ATOMIC_LOAD_U32 | SLJIT_32)
+
+/* Flags: - (does not modify flags) */
+#define SLJIT_ATOMIC_LOAD_U16 (SLJIT_ATOMIC_BASE + 3)
+/* Flags: - (does not modify flags) */
+#define SLJIT_ATOMIC_LOAD32_U16 (SLJIT_ATOMIC_LOAD_U16 | SLJIT_32)
+
+/* Flags: - (does not modify flags) */
+#define SLJIT_ATOMIC_LOAD_U8 (SLJIT_ATOMIC_BASE + 4)
+/* Flags: - (does not modify flags) */
+#define SLJIT_ATOMIC_LOAD32_U8 (SLJIT_ATOMIC_LOAD_U8 | SLJIT_32)
+
+/* Atomic store stores data from one register to memory atomically,
+   clearing a memory lock if necessary and successful.
+   Flags: ZF is cleared if not successful */
+
+/* Flags: Z */
+#define SLJIT_ATOMIC_STORE (SLJIT_ATOMIC_BASE + 5)
+
+/* Flags: Z */
+#define SLJIT_ATOMIC_STORE_U32 (SLJIT_ATOMIC_BASE + 6)
+/* Flags: Z */
+#define SLJIT_ATOMIC_STORE32 (SLJIT_ATOMIC_STORE_U32 | SLJIT_32)
+
+/* Flags: Z */
+#define SLJIT_ATOMIC_STORE_U16 (SLJIT_ATOMIC_BASE + 7)
+/* Flags: Z */
+#define SLJIT_ATOMIC_STORE32_U16 (SLJIT_ATOMIC_STORE_U16 | SLJIT_32)
+
+/* Flags: Z */
+#define SLJIT_ATOMIC_STORE_U8 (SLJIT_ATOMIC_BASE + 8)
+/* Flags: Z */
+#define SLJIT_ATOMIC_STORE32_U8 (SLJIT_ATOMIC_STORE_U8 | SLJIT_32)
+
+SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_emit_atomic_op1(struct sljit_compiler *compiler, sljit_s32 op, sljit_s32 base_reg, sljit_s32 data_reg, sljit_s32 temp_reg);
```

Comment:
We should have two functions, one for load, another for store. This will be useful for archs with more complicated atomic instructions, such as ppc.